backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
   private final UpdateAnnouncementPort updateAnnouncementPort;
+  private final UpdateOrSaveNormPort updateOrSaveNormPort;
+  private final NormService normService;
+  private final TimeMachineService timeMachineService;
+  private final CreateNewVersionOfNormService createNewVersionOfNormService;
+  private final DeleteQueuedNormsPort deleteQueuedNormsPort;
+  private final DeleteUnpublishedNormPort deleteUnpublishedNormPort;
 
   public ReleaseService(
     LoadAnnouncementByNormEliUseCase loadAnnouncementByNormEliUseCase,
-    UpdateAnnouncementPort updateAnnouncementPort
+    UpdateAnnouncementPort updateAnnouncementPort,
+    UpdateOrSaveNormPort updateOrSaveNormPort,
+    NormService normService,
+    TimeMachineService timeMachineService,
+    CreateNewVersionOfNormService createNewVersionOfNormService,
+    DeleteQueuedNormsPort deleteQueuedNormsPort,
+    DeleteUnpublishedNormPort deleteUnpublishedNormPort
   ) {
     this.loadAnnouncementByNormEliUseCase = loadAnnouncementByNormEliUseCase;
     this.updateAnnouncementPort = updateAnnouncementPort;
+    this.updateOrSaveNormPort = updateOrSaveNormPort;
+    this.normService = normService;
+    this.timeMachineService = timeMachineService;
+    this.createNewVersionOfNormService = createNewVersionOfNormService;
+    this.deleteQueuedNormsPort = deleteQueuedNormsPort;
+    this.deleteUnpublishedNormPort = deleteUnpublishedNormPort;
   }
 
+  /**
+   * Queue all norms that are affected by the given announcement for publishing.
+   * This includes generating a new manifestation of the amending norm, and new expressions for each time boundary of
+   * every target norm. It also creates a new manifestation for the current expression of the target norm so it contains
+   * the information about the future changes and the correct "nachfolgende-version-id".
+   * <p></p>
+   * Previous releases of the same announcement of the current date are replaced by this release. Also, new
+   * unpublished manifestations are created for the amending norm and the current expression of the target norms to
+   * enable further work on the announcement.
+   * <p></p>
+   * NOTE: This is currently always creating new expressions for every time boundary (so what is necessary for the
+   * prätext-abgabe). When only metadata has changed to the previous abgabe no new expressions should be generated. This
+   * is not handled by this method yet.
+   *
+   * @param query The query specifying the {@link Announcement} to be loaded.
+   * @return The information about the announcement published.
+   */
   @Override
+  @Transactional
   public Announcement releaseAnnouncement(ReleaseAnnouncementUseCase.Query query) {
     var announcement = loadAnnouncementByNormEliUseCase.loadAnnouncementByNormEli(
       new LoadAnnouncementByNormEliUseCase.Query(query.eli())
     );
-    announcement.setReleasedByDocumentalistAt(Instant.now());
 
+    // find all norms to publish
+    Set<Norm> normsToPublish = new HashSet<>();
```

Comment:
Why do you need a set and then iterate over the set? Won't it always be 1 norm? Since we use the expression eli and the DBService loadNorm is retrieving when passing the expressin eli the latest manifestation right?

Reply:
We also store the target norms in this set so the publishing process can 1. already handle transitive changes and we 2. do not care anymore after this step if a norm is an amending or target norm as we need to do the same steps anyway. (in line 104 the target norms get added to the set)